cobalt/dom/url_media_source.cc

```diff
@@ -32,25 +35,63 @@ void RegisterMediaSourceObjectURL(
     script::EnvironmentSettings* environment_settings,
     const std::string& blob_url,
     const scoped_refptr<dom::MediaSource>& media_source) {
-  dom::DOMSettings* dom_settings =
-      base::polymorphic_downcast<dom::DOMSettings*>(environment_settings);
-  DCHECK(dom_settings);
-  DCHECK(dom_settings->media_source_registry());
+  web::EnvironmentSettings* web_settings =
+      base::polymorphic_downcast<web::EnvironmentSettings*>(
+          environment_settings);
+  DCHECK(web_settings);
+  DCHECK(web_settings->context());
+  web::WindowOrWorkerGlobalScope* global_scope =
+      web_settings->context()->GetWindowOrWorkerGlobalScope();
 
-  scoped_refptr<MediaSourceAttachment> attachment =
-      base::MakeRefCounted<SameThreadMediaSourceAttachment>(media_source);
+  if (global_scope->IsWorker()) {
```

Comment:
Just want to check again, if this is `IsDedicatedWorker` or `IsWorker`.

Reply:
This is checking the settings, which should be agnostic for the type of worker.